mvlearn/preprocessing/preprocessing.py

```diff
@@ -0,0 +1,153 @@
+        shuffle=False then stratify must be None.
+
+    stratify: array-like, default=None
+        If not None, data is split in a stratified fashion, using
+        this as the class labels.
+
+    Returns
+    -------
+    Xs_train : list of array-likes
+        The subset of the data for training.
+
+    Xs_test : list of array-likes
+        The subset of the data for testing.
+
+    y_train : numpy.ndarray or None
+        The subset of the targets for training if y was provided.
+        Otherwise, None is returned.
+
+    y_test : numpy.ndarray or None
+        The subset of the targets for testing if y was provided.
+        Otherwise, None is returned.
+
+    Examples
+    --------
+    >>> from mvlearn.preprocessing import train_test_split
+    >>> import numpy as np
+    >>> RANDOM_STATE=10
+    >>> np.random.seed(RANDOM_STATE)
+    >>> data = np.arange(18).reshape((3, 3, 2))
+    >>> # Print the data
+    >>> for i in range(len(data)):
+    ...     print('View %d' % i, data[i], sep='\n')
+    View 0
+    [[0 1]
+    [2 3]
+    [4 5]]
+    View 1
+    [[ 6  7]
+    [ 8  9]
+    [10 11]]
+    View 2
+    [[12 13]
+    [14 15]
+    [16 17]]
+    >>> train_Xs, test_Xs, _, _ = train_test_split(data, test_size=0.33,
+    ...                                            random_state=RANDOM_STATE)
+    >>> # Printing train set
+    >>> for i in range(len(data)):
+    ...     print('View %d' % i, train_Xs[i], sep='\n')
+    View 0
+    [[4 5]
+    [2 3]]
+    View 1
+    [[10 11]
+    [ 8  9]]
+    View 2
+    [[16 17]
+    [14 15]]
+    # Printing test set
+    >>> for i in range(len(data)):
+    ...     print('View %d' % i, test_Xs[i], sep='\n')
+
+    View 0
+    [[0 1]]
+    View 1
+    [[6 7]]
+    View 2
+    [[12 1]]
+    '''
+
+    if y is None:
+        Xs = check_Xs(Xs)
+        splits = model_selection.train_test_split(*Xs, test_size=test_size,
+                                                  train_size=train_size,
+                                                  random_state=random_state,
+                                                  shuffle=shuffle,
+                                                  stratify=stratify)
+    else:
+        Xs, y = check_Xs_y(Xs, y)
+        splits = model_selection.train_test_split(*Xs, y, test_size=test_size,
+                                                  train_size=train_size,
+                                                  random_state=random_state,
+                                                  shuffle=shuffle,
+                                                  stratify=stratify)
+
+    Xs_train = list()
+    Xs_test = list()
+    for i in range(len(splits)):
+        if i % 2 == 0:
+            Xs_train.append(splits[i])
+        else:
+            Xs_test.append(splits[i])
+
+    y_train = y_test = None
+
+    if y is not None:
+        y_train = Xs_train.pop()
+        y_test = Xs_test.pop()
+
+    return Xs_train, Xs_test, y_train, y_test
```

Comment:
Doesn't sklearn return a list of length 2*n_inputs? Maybe make it a list, though I see how then it becomes a list of lists of arrays lol

Also, I think rather than returning None for the y_train and y_test, they should only be returned if they are passed in.

What do you think?

Reply:
yeah I wouldn't return none, because then it's a sort of weird 2 input 4 output situation. It's unintuitive definitely.